api/layer/tagging.go

```diff
@@ -10,41 +10,44 @@ import (
 	"go.uber.org/zap"
 )
 
-func (n *layer) GetObjectTagging(ctx context.Context, p *ObjectVersion) (map[string]string, error) {
+func (n *layer) GetObjectTagging(ctx context.Context, p *ObjectVersion) (string, map[string]string, error) {
 	var (
 		err  error
 		tags map[string]string
 	)
+
 	tags = n.systemCache.GetTagging(objectTaggingCacheKey(p))
 	if tags != nil {
-		return tags, nil
+		return p.VersionID, tags, nil
 	}
 
 	version, err := n.getNodeVersion(ctx, p)
 	if err != nil {
-		return nil, err
+		return "", nil, err
 	}
+	p.VersionID = version.OID.EncodeToString()
```

Comment:
We don't use `p.Version` after the assignment

Reply:
In line 38, we form key to cache tags